simulation/sub8_simulation/nodes/simulate.py

```diff
@@ -55,21 +59,65 @@ def add_box(self, position, density, width, height, depth, color):
         self.entity_pairs.append((visual, physical))
 
     def add_sub(self, position):
+        '''
+        TODO:
+            This should be some kind of seperate widget, but alas, it is currently just jammed in here
+        '''
         visual = self.rendering_world.add_mesh(Sub8Visual, position,
                                       orientation=None, color=(20, 20, 20, 1), shininess=20)
         physical = self.physics_world.add_entity(Sub8, position)
-        self.rendering_world.add_entity(rendering.Indicator, physical, radius=0.2)
-
-        self.rendering_world.add_entity(rendering.Indicator, physical, get_param=lambda o: np.array(o.last_force), 
-                                        radius=0.1, color=(200, 10, 0), scaling_factor=0.01)
+        self.rendering_world.add_entity(rendering.Indicator, physical, radius=0.05)
+
+        self.rendering_world.add_entity(rendering.Indicator, physical, get_param=lambda o: np.array(o.last_force),
+                                        radius=0.025, color=(200, 10, 0), scaling_factor=0.01)
+
+        thruster_list = [
+            ("FLV", np.array([ 0.000,  0.0, -1]), np.array([ 0.1583, 0.16900, 0.0142])),
+            ("FLL", np.array([-0.866,  0.5,  0]), np.array([ 0.2678, 0.27950, 0.0000])),
+            ("FRV", np.array([ 0.000,  0.0, -1]), np.array([ 0.1583, -0.1690, 0.0142])),
+            ("FRL", np.array([-0.866, -0.5,  0]), np.array([ 0.2678, -0.2795, 0.0000])),
+            ("BLV", np.array([ 0.000,  0.0,  1]), np.array([-0.1583, 0.16900, 0.0142])),
+            ("BLL", np.array([ 0.866,  0.5,  0]), np.array([-0.2678, 0.27950, 0.0000])),
+            ("BRV", np.array([ 0.000,  0.0,  1]), np.array([-0.1583, -0.1690, 0.0142])),
+            ("BRL", np.array([ 0.866, -0.5,  0]), np.array([-0.2678, -0.2795, 0.0000])),
+        ]
+
+        # add the thrust indicators
+        for name, rel_direction, rel_position in thruster_list:
+
+            R = make_rotation(np.array([0.0, 0.0, 1.0]), rel_direction)
+            transformation = compose_transformation(np.transpose(R), rel_position)
+
+            class ThrustGetter(object):
```

Comment:
This is crappy right now - in the next iteration, I will move this into a "widget" in a seperate file